Userland/Libraries/LibJS/Tests/builtins/String/String.prototype.localeCompare.js

```diff
@@ -0,0 +1,30 @@
+test("basic functionality", () => {
+    expect(String.prototype.localeCompare).toHaveLength(1);
+
+    expect("".localeCompare("")).toBe(0);
+    expect("a".localeCompare("a")).toBe(0);
+    expect("6".localeCompare("6")).toBe(0);
+
+    expect("".localeCompare("a")).not.toBe(0);
+    expect("a".localeCompare("b")).not.toBe(0);
+    expect("6".localeCompare("7")).not.toBe(0);
+    expect("a".localeCompare("A")).not.toBe(0);
+
+    expect("undefined".localeCompare()).toBe(0);
+    expect("undefined".localeCompare(undefined)).toBe(0);
+
+    expect("null".localeCompare(null)).toBe(0);
+    expect("null".localeCompare(undefined)).not.toBe(0);
+    expect("null".localeCompare()).not.toBe(0);
+
+    expect(() => {
+        String.prototype.localeCompare.call(undefined, undefined);
+    }).toThrow(TypeError);
```

Comment:
Please include messages in error testing (`toThrowWithMessage`), unless there's a good reason not to (which isn't the case here)